add python matter server

hgy59 opened the pull request:
## Description

- initial version of python-matter-server with custom chip library for home-assistant-chip-core

Closes #6321

## Restrictions

findings so far
- Wireless networks are not supported. Synology DSM does not support WLAN Adapters since DSM 6.2 and DSM 7.
- Ethernet Adapter on VirtualDSM is not found by matter-server (might be related to virtio_net driver)

## Checklist

- [x] Build rule `all-supported` completed successfully
- [x] New installation of package completed successfully
- [x] Package upgrade completed successfully (Manually install the package again)
- [ ] Package [functionality was tested](https://github.com/SynoCommunity/spksrc/wiki/Package-Update-Policy#tests-checks)
- [ ] Any needed [documentation](https://github.com/SynoCommunity/spksrc/wiki/Create-documentation) is updated/created


### Type of change

<!--Please use any relevant tags.-->
- [x] New Package

hgy59 committed:
add python matter server

- initial version with custom chip library for home-assistant-chip-core

hgy59 commented:
@24mu13 I found a workaround to avoid `/data` folder for chip library config files.

The custom libraries are taken from https://github.com/hgy59/chip-wheels
The https://pypi.org/project/home-assistant-chip-core/ is a dependency of matter-server and has hard coded `/data` path for chip configuration files.
In the branch [feature-custom-config-data-path](https://github.com/hgy59/chip-wheels/tree/feature-custom-config-data-path) I added a patch to allow custom config data path defined by env variable.

If I will get confirmations that this matter-server package works, I probably will create a PR in https://github.com/home-assistant-libs/chip-wheels to officially support custom config data path.
So far the HA matter components are designed to run in docker containers where it is no issue to store all config data in `/data`.
With the workaround it is possible to store such data under the package var folder.

Since I do not have matter compatible devices for functional tests, every feedback to this packag is welcome.

hgy59 committed:
build python-matter-server wheel depending on custom home_assistant_chip_core

- avoid need of chip library replacement of original home_assistant_chip_core wheel
- build pure python-matter-server wheel depending on custom home_assistant_chip_core wheel
- patch python-matter-server source to custom wheel and to force original version in check
- install custom python-matter-server and home_assistant_chip_core version 2025.3.dev1801
- add matter-server admin page

hgy59 committed:
use custom home_assistant_chip_core wheel with original version number

- avoid need to create patched python-matter-server wheel to use custom wheel

hgy59 committed:
Merge branch 'master' of https://github.com/hgy59/spksrc into add_matter-server

hgy59 commented:
@24mu13 I found a better solution than replacing the lib file at installation. I now built the custon wheel with original version number (and additionally stripped the library file) so it perfectly works together with oribinal python-matter-server wheel.

I also added a link to the admin page.

hgy59 committed:
matter-server: adjust comments in requirements files

hgy59 commented:
@24mu13 are you still looking for a package of python-matter-server?

hgy59 committed:
Merge branch 'master' of https://github.com/hgy59/spksrc into add_matter-server

hgy59 committed:
matter-server: update to v8.0.0

hgy59 committed:
adjust homepage and license

hgy59 commented:
@24mu13 are you still looking for matter server package?

I am waiting for confirmation that it works.

hgy59 committed:
Merge branch 'master' of https://github.com/hgy59/spksrc into add_matter-server

24mu13 commented:
> I guess you have seen that Supervised HA and HA Core will not be supported anymore with HA 2025.12?
> https://www.home-assistant.io/blog/2025/05/22/deprecating-core-and-supervised-installation-methods-and-32-bit-systems/

Thank you for the info: this will not affect me (I have DS224+ product with 64bit architecture) but you're right - all 32bit architectures instead are affected by this deprecation.

hgy59 commented:
I have created pre-release versions of matter-server 7.0.1 and 8.0.0
https://github.com/hgy59/spksrc/releases/tag/matter-server

hgy59 commented:
> I did however note in the UI that the icon in the Main Menu is missing and the open button in the Package Center is not active.

That is by intention
I didn't define SERVICE_PORT, only ADMIN_PORT is defined.
So there is a link in the Package Center and the UI is available for admins only.

We have a lot of packages that define ADMIN_PORT = $(SERVICE_PORT), but IMHO this is not good practice. We should use ADMIN_PORT only if it is different to SERVICE_PORT i.e. the regular UI of a package is under SERVICE_PORT and (a different) admin UI is on a different port.
Or we can define ADMIN_PORT _instead of_ SERVICE_PORT to avoid an icon (link) in the DSM UI (as I did here).

IMHO we shouldn't fix this in existing packages, because users might be used to use the admin link in package center,
but for new Packages this is not the case.

mreid-tt commented:
> That is by intention
> I didn't define SERVICE_PORT, only ADMIN_PORT is defined.
> So there is a link in the Package Center and the UI is available for admins only.

So you’re suggesting installing a package without an easy-to-access admin interface? Is that because it lacks authentication and you’re concerned about inexperienced users making changes? Could you share another package where you’d recommend a similar approach? I’m trying to put this in context and better understand the intent.

hgy59 commented:
> Thank you for the info: this will not affect me

But it will affect your Supervised installation, as supervised will get deprecated.
And we can't create further HA packages, because Home Assistant Core gets deprecated too.

On Synology Devices, only Docker installations (or may be Home Assistant OS as virtual machine) will be supported.

hgy59 commented:
> So you’re suggesting installing a package without an easy-to-access admin interface?

I thought that everyone used the url to open the packages, because of the definition of ADMIN_PORT.
If you want to open the packages with the open button or with the package icon in DSM UI, we wouldn't need the URL in Package Center (i.e. the definition of ADMIN_PORT).

I want to point out, that there is a difference between package interface and admin interface..
For the package interface you have the open button in package center and the package icon in DSM UI.
For the admin interface you have the URL in Package Center.
For packages with SERVICE_PORT you define the ADMIN_PORT for the admin interface.
For web packages with a web service config, you define ADMIN_URL and ADMIN_PROTOCOL for the admin interface.

If both are the same, it does not make sense to have the URL in package center.


For me the matter server package web ui is not a common web ui (like in owncloud) it is only to administrate matter devices.
It is easy to access with the URL in package center.

hgy59 committed:
matter-server: update to v8.1.0

- add chip_core wheels 2025.7.0
- update requirements
- install package with integration into DSM UI

hgy59 commented:
@mreid-tt I have now switched to SERVICE_PORT (and DSM UI integration).
The first package release didn't have neither SERVICE_PROT nor ADMIN_PORT defined, as it is mainly a webservice interface.
Later I found that it also presents the admin interface on this port.

mreid-tt commented:
@hgy59 this is a very different understanding than what I had for its use. Looking into our documentation we have:

Variable name | Explanation | Example
-- | -- | --
ADMIN_PORT | Port for web based management interface of package. Used to create a link in DSM package center. Required only if SERVICE_PORT is not defined or different. Another use case is to define ADMIN_PORT = $(SERVICE_PORT) together with NO_SERVICE_SHORTCUT = true to get the web interface in the DSM package center but not in the DSM UI. | ADMIN_PORT = 9091
SERVICE_PORT | TCP port of generic service support for firewall and DSM UI shortcut. For new packages please choose a free port and update the list SynoCommunity Used Ports | SERVICE_PORT = 8080

This is in alignment with what you have stated and in the DSM Developer Guide we have definitions of the admin port as follows:

### Field Name: adminport
* Description: A package listens to a specific port to display its own UI. If the package is defined by a port, a link will be opened when the package is started.
> Note: adminprotocol, adminportand adminurl keys are combined to adminprotocol://ip:adminport/adminurl link

This as noted, is separate from the UI files which define the icon and button behaviour for the app.

The issue, as I see it, is that we’ve blended the two for so long that this may now be the expected behavior for our users.

A quick check of the Makefiles in spksrc shows 58 files with an `ADMIN_PORT` defined. Of these, only 13 — mostly less common apps such as darkstat, ejabberd, go2rtc, haproxy, htpcmanager, maraschino, minio, monit, node-exporter, nzbmegasearch, plexivity, rabbitmq, and saltpad — use something other than `ADMIN_PORT = $(SERVICE_PORT)`.

Since I haven’t contributed code to these particular apps, I’ve never encountered a case where the *Open* button and the admin URL pointed to different places.

hgy59 commented:
@mreid-tt the icecast package has such an approach.
It has ADMIN_PORT = $(SERVICE_PORT) but also SERVICE_URL = /admin/
so the Open button and DSM UI icon will open http://{nas-ip}:8000
and the URL in package center will open http://{nas-ip}:8000/admin/
since /admin/ path is also available in the regular web ui, it is not mandatory to have it.

A real use case would be, when the Open Button opens a Dialog integrated into DSM and the admin url opens a web UI of the package.
But we only had such Integration based on python2 code, and that is deprecated long ago (only DSM 5 really worked, like in debian-chroot package) and none of those had an additional web interface (AFAICR) ....

I propose to avoid ADMIN_PORT, ADMIN_URL, SERVICE_URL, etc. at least when the package does not provide two different web interfaces.

mreid-tt commented:
> I propose to avoid ADMIN_PORT, ADMIN_URL, SERVICE_URL, etc. at least when the package does not provide two different web interfaces.

Agreed. I'll keep this in mind for any future code I work on.

hgy59 committed:
Merge branch 'master' of https://github.com/hgy59/spksrc into add_matter-server

hgy59 committed:
matter-server: update to v8.1.1

- update requirements
- use specific crossenv for yarl 1.22.0

hgy59 commented:
@24mu13 Thanks for the feedback.
It is now updated to v8.1.1.

hgy59 committed:
Merge branch 'master' of https://github.com/hgy59/spksrc into add_matter-server

hgy59 committed:
matter-server: update to v8.1.2

- update requirements
- redesign to avoid to build cross wheels (for x64 and aarch64 wheels are available in the index)
- all wheels (except the prebuilt home_assistant_chip_core) are downloaded at installation time
- no need of python to build the package anymore

hgy59 committed:
matter-server: cleanup

- remove obsolete file
